subsys/net/lib/download_client/src/coap.c

```diff
@@ -100,6 +141,10 @@ int coap_parse(struct download_client *client, size_t len)
 	const uint8_t *payload;
 	struct coap_packet response;
 
+	/* TODO: currently we stop download on every error, but this is mostly not necessary
```

Comment:
Is that still the case? Looking at the logic below, we go directly to `send_again` label in case of timeout (provided we still have retransmissions left. IMO it's reasonable to reconnect in any other case (especially for DTLS).

Reply:
If coap_packet_parse fails, we drop out of the loop and go to "restart_and_suspend". Please see:

```
[00:00:26.396,606] <err> download_client: Failed to parse CoAP packet, err 1
[00:00:26.397,186] <err> fota_download: Download client error
[00:00:26.397,186] <inf> dfu_target_mcuboot: MCUBoot image upgrade aborted.
[00:00:26.397,186] <err> lwm2m_firmware: FOTA_DOWNLOAD_EVT_ERROR
```

Reply:
Ok, scratch that, I assumed the TODO refers to changes in this PR (socket-related errors).